waiter/src/waiter/process_request.clj

```diff
@@ -544,40 +526,49 @@
+                                                                  start-new-service-fn request-state-chan queue-timeout-ms
+                                                                  reservation-status-promise metric-group)))
+                        instance (:out timed-instance)
+                        instance-elapsed (:elapsed timed-instance)]
+                    (statsd/histo! metric-group "get_instance" instance-elapsed)
                     (-> (try
                           (log/info "suggested instance:" (:id instance) (:host instance) (:port instance))
                           (confirm-live-connection-without-abort)
                           (let [endpoint (request->endpoint request waiter-headers)
-                                {:keys [error] :as response}
-                                (metrics/with-timer!
-                                  (metrics/service-timer service-id "backend-response")
-                                  (fn [nanos]
-                                    (add-debug-header-into-response! "X-Waiter-Backend-Response-ns" nanos)
-                                    (statsd/histo! metric-group "backend_response" nanos))
-                                  (async/<!
-                                    (make-request-fn instance request instance-request-properties
-                                                     passthrough-headers endpoint metric-group)))
+                                backend-timer (metrics/service-timer service-id "backend-response")
+                                timed-response (metrics/with-timer
+                                                 backend-timer
+                                                 (async/<!
+                                                   (make-request-fn instance request instance-request-properties
+                                                                    passthrough-headers endpoint metric-group)))
+                                response-elapsed (:elapsed timed-response)
+                                {:keys [error] :as response} (:out timed-response)
                                 request-abort-callback (request-abort-callback-factory response)
                                 confirm-live-connection-with-abort (confirm-live-connection-factory request-abort-callback)]
-                            (when error
-                              (throw-response-error error reservation-status-promise instance response-headers))
-                            (process-backend-response-fn local-usage-agent instance-request-properties descriptor instance request
-                                                         reason-map response-headers reservation-status-promise
-                                                         confirm-live-connection-with-abort request-state-chan response))
+                            (statsd/histo! metric-group "backend_response" response-elapsed)
```

Comment:
Observation: working with `response-elapsed` is no longer in spatially close locations, unlike the previous version.